### PR TITLE
[DEV APPROVED] 7127 add warning to retirement assets

### PR DESCRIPTION
**Pull request by Tr4pSt3R**
[DO NOT MEGE] until this PR(https://github.com/moneyadviceservice/frontend/pull/1423) gets through

---
This change is [Reviewable](https://reviewable.io/reviews/moneyadviceservice/frontend/1425)

### CODE REVIEW

File: spec/helpers/budget_warning_spec.rb [hunk 1]
```diff
@@ -21,26 +21,83 @@
...
+                                  base_url: 'www.example.com')
+         allow(BudgetWarning).to receive(:request).and_return(request)
+
+         expect(described_class.whitelisted?).to be(true)
+       end
+
+       it "displays warning banner for income-drawdown" do
+          request = double('request', url: "www.example.com/en/retirement-income-options/income-drawdown",
+                                   base_url: 'www.example.com')
+          allow(BudgetWarning).to receive(:request).and_return(request)
```
Comment (urfolomeus) [inline spec/helpers/budget_warning_spec.rb]: Seems like this is common setup. Worth extracting?

**Comment by urfolomeus** [comment]
Nice programming :+1: 😁

File: spec/helpers/budget_warning_spec.rb (hunk 1 above)
Comment (Tr4pSt3R) [inline spec/helpers/budget_warning_spec.rb]: yes, well spotted! @urfolomeus

**Comment by rnalexander** [comment]
:+1:

**Comment by Tr4pSt3R** [comment]
Hi @urfolomeus, per your recommendation I have stripped down the tests and stuffed them in a shared example file thereby hiding unnecessary lines of code from the spec file. 

I was on holiday. I have picked this up again Today. Have a look and add more comments please @rnalexander

**Comment by urfolomeus** [comment]
Awesome! :) :+1: :100: :tada:

**Comment by Tr4pSt3R** [comment]
Thanks @urfolomeus :D :dancer:

**Comment by rnalexander** [comment]
:+1: :shipit:

**Comment by Tr4pSt3R** [comment]
Thanks @rnalexander && @urfolomeus :+1:

**Comment by tomas-stefano** [comment]
Very good mate. :+1:

**Pull request merged**